### PR TITLE
docs(webpack): add custom application and activity section

### PR DESCRIPTION
**Pull request by vchimev**
## PR Checklist

- [x] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md#commit-messages.
- [ ] If there is an issue related with this PR, point it out here.

## What is the new state of the documentation article?
Add custom application and activity configuration information.

Describes https://github.com/NativeScript/nativescript-dev-webpack/issues/564#issuecomment-399675425.

### CODE REVIEW

File: docs/performance-optimizations/bundling-with-webpack.md [hunk 1]
```diff
@@ -209,6 +209,41 @@ function logMessage(message) {
 }
 ```
 
+## Custom Application and Activity
+
+NativeScript provides a way to create custom `android.app.Application` and `android.app.Activity` implementations described in [this](https://docs.nativescript.org/core-concepts/android-runtime/advanced-topics/extend-application-activity) documentation article. In order to bundle such an application, it needs to be configured.
+
+### Extend Android Application
+
+If your project extends android application, it should be added as an [entry](https://github.com/NativeScript/nativescript-dev-webpack/blob/master/demo/AngularApp/webpack.config.js#L71) to the `webpack.config.js` file.
```
Comment (etabakov) [inline docs/performance-optimizations/bundling-with-webpack.md]: - "extends android application" should be "extends an Android application"

File: docs/performance-optimizations/bundling-with-webpack.md [hunk 2]
```diff
@@ -209,6 +209,41 @@ function logMessage(message) {
...
+```
+entry: {
+    bundle: entryPath,
+    application: "./application.android",
+},
+```
+
+In this way, the source code of [`application.android.ts`](https://github.com/NativeScript/nativescript-dev-webpack/blob/master/demo/AngularApp/app/application.android.ts) is bundled separately as `application.js` file which is loaded from the native `Application.java` class on launch.
+
+The `application.js` bundle file is independent of the `bundle.js` and `vendor.js` files and the reason for it is that they are not loaded so early in the application launch. That's why the logic in `application.android.ts` is needed to be bundled separately in order to be loaded by the native `Application.java` as early as needed on launch.
```
Comment (etabakov) [inline docs/performance-optimizations/bundling-with-webpack.md]: - "the reason for it" would sound better as "the reason for this/that"

File: docs/performance-optimizations/bundling-with-webpack.md [hunk 3]
```diff
@@ -209,6 +209,41 @@ function logMessage(message) {
...
+
+> Note: This approach won't work if `aplication.android.ts` requires external modules.
+
+### Extend Android Activity
+
+If your project extends android activity, it should be added to the [`appComponents`](https://github.com/NativeScript/nativescript-dev-webpack/blob/master/demo/AngularApp/webpack.config.js#L19) array by an absolute path.
```
Comment (etabakov) [inline docs/performance-optimizations/bundling-with-webpack.md]: - again "extends an Android activity"

File: docs/performance-optimizations/bundling-with-webpack.md [hunk 4]
```diff
@@ -209,6 +209,41 @@ function logMessage(message) {
...
+
+```
+const appComponents = [
+    "tns-core-modules/ui/frame",
+    "tns-core-modules/ui/frame/activity",
+    resolve(__dirname, "app/activity.android.ts"),
+];
+```
+
+In this way and With the default config, these components [get](https://github.com/NativeScript/nativescript-dev-webpack/blob/master/demo/AngularApp/webpack.config.js#L109-L116) in the common *vendor.js* chunk and are required by the [`android-app-comopnents-loader`](https://github.com/NativeScript/nativescript-dev-webpack/blob/master/demo/AngularApp/webpack.config.js#L146-L149).
```
Comment (etabakov) [inline docs/performance-optimizations/bundling-with-webpack.md]: - not sure why "With" is capitalized
 - a typo in "android-app-comopnents-loader"

**Review by etabakov — COMMENTED** [review]
Some minor issues ...

**Review by sis0k0 — CHANGES_REQUESTED** [review]
I think these sections should be part of the docs article for extending application and activity. The webpack article can link to the sections in the other article.

**Review by sis0k0 — APPROVED** [review]

**Pull request merged**